force-app/main/default/classes/SOQL.cls

```diff
@@ -130,232 +249,541 @@ public with sharing class SOQL {
      * @return A list of Opportunities with an Amount greater than 100,000, ordered by CloseDate, and limited to 5 records.
      */
     public static List<Opportunity> getHighValueOpportunities() {
-        return null; // Replace null with the variable you used to store the result
+        // Check for Opportunity read permissions
+        if (!Schema.sObjectType.Opportunity.isAccessible()){
+            throw new OtherException('Current user does not have permission to read Opportunity records.');
+        }   
+
+        List<Opportunity> opportunityList = [
+            SELECT
+                Id,
+                Name,
+                AccountId,
+                CloseDate,
+                StageName,
+                Amount
+            FROM
+                Opportunity
+            WHERE
+                Amount > 100000
+            ORDER BY
+                CloseDate
+            LIMIT
+                5
+        ];
+
+        return opportunityList; // Replace null with the variable you used to store the result
     }
     
     /**
-     * Question 15
+     * Question 15 Completed
      * Retrieves the Contacts with the most recent Birthdate, ordered by Birthdate, and limited to 3 records.
      * @return A list of Contacts
      */
     public static List<Contact> getBirthdateContacts() {
-        return null; // Replace null with the variable you used to store the result
+        // Check for Contact read permissions
+        if (!Schema.sObjectType.Contact.isAccessible()){
+            throw new OtherException('Current user does not have permission to read Contact records.');
+        }
+
+            List<Contact> contactList = [
+            SELECT
+                Id,
+                LastName,
+                Birthdate
+            FROM
+                Contact
+            ORDER BY
+                Birthdate DESC
+            LIMIT
+                3
+        ];
+
+        return contactList; // Replace null with the variable you used to store the result
     }
     
     /**
-     * Question 16
+     * Question 16 Completed
      * Retrieves the Cases in the Open Status, ordered by CreatedDate, and limited to 10 records with an offset of 5.
      * @return A list of Cases
      */
     public static List<Case> getCasesWithStatus() {
-        return null; // Replace null with the variable you used to store the result
+        // Check for Case read permissions
+        if (!Schema.sObjectType.Case.isAccessible()){
+            throw new OtherException('Current user does not have permission to read Case records.');
+        }
+
+        List<Case> caseList = [
+            SELECT
+                Id,
+                Subject,
+                Status,
+                CreatedDate
+            FROM
+                Case
+            WHERE
+                Status = 'Open'
+            ORDER BY
+                CreatedDate
+            LIMIT
+                10
+            OFFSET
+                5
```

Comment:
Personal preference: I'm not a fan of this formatting for SOQL, but overall, it's fine.